network/debinterfaces/bridge_test.go

```diff
@@ -0,0 +1,326 @@
+
+func (s *BridgeSuite) TestBridgeBond(c *gc.C) {
+	input := `
+auto bond0
+iface bond0 inet static
+    address 10.17.20.211/24
+    gateway 10.17.20.1
+    dns-nameservers 10.17.20.200
+    bond-slaves none
+    bond-mode active-backup
+    bond-xmit_hash_policy layer2
+    bond-miimon 100
+    mtu 1500
+    hwaddress 52:54:00:1c:f1:5b
+    bond-lacp_rate slow`
+
+	expected := `
+auto bond0
+iface bond0 inet manual
+    bond-slaves none
+    bond-mode active-backup
+    bond-xmit_hash_policy layer2
+    bond-miimon 100
+    mtu 1500
+    hwaddress 52:54:00:1c:f1:5b
+    bond-lacp_rate slow
+
+auto br-bond0
+iface br-bond0 inet static
+    address 10.17.20.211/24
+    gateway 10.17.20.1
+    dns-nameservers 10.17.20.200
+    hwaddress 52:54:00:1c:f1:5b
+    bridge_ports bond0`
+	s.assertBridge(input, expected[1:], c, map[string]string{"bond0": "br-bond0"})
+}
+
+func (s *BridgeSuite) TestBridgeNoIfacesDefined(c *gc.C) {
+	input := `
+mapping eth0
+    script /path/to/pcmcia-compat.sh
+    map home,*,*,*                  home
+    map work,*,*,00:11:22:33:44:55  work-wireless
+    map work,*,*,01:12:23:34:45:50  work-static`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeBondMaster(c *gc.C) {
+	input := `
+auto ens5
+iface ens5 inet manual
+    bond-lacp_rate slow
+    mtu 1500
+    bond-master bond0
+    bond-xmit_hash_policy layer2
+    bond-mode active-backup
+    bond-miimon 100`
+	s.assertBridge(input, input[1:], c, map[string]string{"eth0": "br-eth0"})
+}
+
+func (s *BridgeSuite) TestBridgeNoIfacesDefinedFromFile(c *gc.C) {
+	stanzas, err := debinterfaces.ParseSource("testdata/ifupdown-examples", nil, s.expander)
+	c.Assert(err, gc.IsNil)
+	input := format(stanzas)
+	s.assertBridge(input, input, c, map[string]string{"non-existent-interface": "non-existent-bridge"})
+}
+
+func (s *BridgeSuite) TestBridgeAlias(c *gc.C) {
+	input := `
+auto eth0
+iface eth0 inet static
+    gateway 10.14.0.1
+    address 10.14.0.102/24
+
+auto eth0:1
+iface eth0:1 inet static
+    address 1.2.3.5`
+
+	expected := `
+auto eth0
+iface eth0 inet manual
+
+auto br-eth0:1
+iface br-eth0:1 inet static
+    address 1.2.3.5
+
+auto br-eth0
+iface br-eth0 inet static
+    gateway 10.14.0.1
+    address 10.14.0.102/24
+    bridge_ports eth0`
+	s.assertBridge(input, expected[1:], c, map[string]string{"eth0": "br-eth0", "eth0:1": "br-eth0:1"})
+}
+
+func (s *BridgeSuite) TestBridgeMultipleInterfaces(c *gc.C) {
+	input := `
+auto enp1s0f3
+iface enp1s0f3 inet static
+  address 192.168.1.64/24
+  gateway 192.168.1.254
+  dns-nameservers 192.168.1.254
+  dns-search home
+
+iface enp1s0f3 inet6 dhcp`
+
+	expected := `
+auto enp1s0f3
+iface enp1s0f3 inet manual
+
+iface enp1s0f3 inet6 manual
+
+auto br-enp1s0f3
+iface br-enp1s0f3 inet static
+    address 192.168.1.64/24
+    gateway 192.168.1.254
+    dns-nameservers 192.168.1.254
+    dns-search home
+    bridge_ports enp1s0f3
+
+iface br-enp1s0f3 inet6 dhcp
```

Comment:
I don't think it breaks anything to have bridge_ports in both, and otherwise we'd have to keep track whether we added them (we can have an inet6 only iface).